<a href="https://colab.research.google.com/github/ashlearo/ML/blob/main/%D0%95%D1%80%D0%BC%D0%B0%D0%BA%D0%BE%D0%B2_%D0%94%D0%97_5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np

In [ ]:
from google.colab import files
uploaded = files.upload()

Saving job_posting_small.csv to job_posting_small.csv


In [ ]:
job_posting_small = pd.read_csv('job_posting_small.csv')
job_posting_small.head()

job_id  ... fraudulent
0       1  ...          0
1       2  ...          0
2       3  ...          0
3       4  ...          0
4       5  ...          0

[5 rows x 18 columns]

In [ ]:
job_posting_small.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 500 entries, 0 to 499
Data columns (total 18 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   job_id               500 non-null    int64 
 1   title                500 non-null    object
 2   location             491 non-null    object
 3   department           152 non-null    object
 4   salary_range         82 non-null     object
 5   company_profile      430 non-null    object
 6   description          500 non-null    object
 7   requirements         443 non-null    object
 8   benefits             290 non-null    object
 9   telecommuting        500 non-null    int64 
 10  has_company_logo     500 non-null    int64 
 11  has_questions        500 non-null    int64 
 12  employment_type      423 non-null    object
 13  required_experience  313 non-null    object
 14  required_education   336 non-null    object
 15  industry             389 non-null    object
 16  function

в таблице 500 строк ,
" 8   benefits             290 non-null    object "
=>    500-290 =  210   пропусков

4   salary_range         82 non-null     object   =>  418 пропусков

In [ ]:
job_posting_small.groupby(['department']).describe()

job_id                                 ... fraudulent               
                    count        mean         std    min  ...        25%  50%  75%  max
department                                                ...                          
 R&D                  1.0   17.000000         NaN   17.0  ...        0.0  0.0  0.0  0.0
AFSDTV                1.0  479.000000         NaN  479.0  ...        0.0  0.0  0.0  0.0
ANDROIDPIT            1.0    7.000000         NaN    7.0  ...        0.0  0.0  0.0  0.0
Account Management    1.0  341.000000         NaN  341.0  ...        0.0  0.0  0.0  0.0
Accounting            1.0  431.000000         NaN  431.0  ...        0.0  0.0  0.0  0.0
...                   ...         ...         ...    ...  ...        ...  ...  ...  ...
Technology            3.0  328.333333  142.514327  172.0  ...        0.0  0.0  0.0  0.0
hr                    1.0  358.000000         NaN  358.0  ...        1.0  1.0  1.0  1.0
non-tech              2.0  429.000000   39.597980  401.0  ...        0.0  0.0  0.0  0.0
sales                 1.0  436.000000         NaN  436.0  ...        0.0  0.0  0.0  0.0
tech                  5.0  234.600000   83.122199  122.0  ...        0.0  0.0  0.0  0.0

[84 rows x 40 columns]

In [ ]:
job_posting_small.department.value_counts()   # интересно сделать словарь:  department - количество

Marketing                         13
Sales                             13
Engineering                        7
IT                                 6
tech                               5
                                  ..
Cast Partial                       1
ICM                                1
Sales and Business Development     1
Customer Care                      1
Design                             1
Name: department, Length: 84, dtype: int64

In [ ]:
job_posting_small['department'].tolist()

['Marketing',
 'Success',
 nan,
 'Sales',
 nan,
 nan,
 'ANDROIDPIT',
 nan,
 nan,
 nan,
 nan,
 'HR',
 nan,
 nan,
 'Sales',
 'Sales',
 ' R&D',
 nan,
 nan,
 nan,
 nan,
 nan,
 'Engagement',
 'Businessfriend.com',
 nan,
 nan,
 'Marketing',
 'Medical',
 nan,
 nan,
 nan,
 nan,
 'Field',
 nan,
 'All',
 nan,
 nan,
 nan,
 'Design',
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 'Production',
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 'ICM',
 nan,
 nan,
 nan,
 'General Services',
 nan,
 nan,
 nan,
 nan,
 'Engineering',
 nan,
 nan,
 'IT',
 'Marketing',
 nan,
 nan,
 nan,
 nan,
 'Engineering',
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 'Business Development',
 nan,
 'Marketing',
 nan,
 'Human Resources',
 nan,
 nan,
 'Marketing',
 nan,
 nan,
 'Oil & Energy',
 nan,
 nan,
 nan,
 'Marketplace',
 nan,
 nan,
 'Cloud Services',
 nan,
 'FP',
 'Client Services',
 nan,
 nan,
 nan,
 nan,
 'Operations',
 'Sales',
 nan,
 nan,
 'Materials',
 nan,
 nan,
 nan,
 'tech',
 

In [ ]:

req_ex = job_posting_small.groupby([ 'required_experience' , 'required_education'])
req_ex.describe()

#   по значениям count  видим, что макс High School or equivalent в Entry level
#   не очень наглядно, интересно, как это можно сделать другими способами

job_id  ... fraudulent
                                                       count  ...        max
required_experience required_education                        ...           
Associate           Associate Degree                     1.0  ...        0.0
                    Bachelor's Degree                   30.0  ...        0.0
                    Certification                        1.0  ...        0.0
                    High School or equivalent            8.0  ...        0.0
                    Master's Degree                      1.0  ...        0.0
                    Unspecified                          7.0  ...        0.0
                    Vocational                           1.0  ...        0.0
Director            Associate Degree                     1.0  ...        0.0
                    Bachelor's Degree                    7.0  ...        0.0
                    Master's Degree                      2.0  ...        0.0
Entry level         Bachelor's Degree                   22.0  ...        0.0
                    High School or equivalent           79.0  ...        1.0
                    Some College Coursework Completed    1.0  ...        0.0
                    Unspecified                          5.0  ...        0.0
Executive           Bachelor's Degree                    1.0  ...        0.0
                    Unspecified                          1.0  ...        0.0
Internship          Bachelor's Degree                    3.0  ...        0.0
                    High School or equivalent            2.0  ...        0.0
                    Master's Degree                      1.0  ...        0.0
                    Some College Coursework Completed    1.0  ...        0.0
Mid-Senior level    Bachelor's Degree                   51.0  ...        0.0
                    Certification                        2.0  ...        1.0
                    High School or equivalent            4.0  ...        1.0
                    Master's Degree                      5.0  ...        0.0
                    Unspecified                          5.0  ...        0.0
Not Applicable      Bachelor's Degree                    1.0  ...        0.0
                    High School or equivalent           11.0  ...        0.0
                    Master's Degree                      1.0  ...        0.0
                    Unspecified                         12.0  ...        0.0

[29 rows x 40 columns]

In [ ]:
# Далее эксперименты с вопросами
# попробовал посмотреть эти зависимости через сводную тапблицу и результат совсем сбил с правильного пути,
# пока ниже не запустил код  job_posting_small[job_posting_small['required_education'].isin(['Associate Degree'])]
# что однозначно определило job_id  - 	 как номер строки
# интересно как посмотреть в сводной таблице количества уникальных значений  'required_experience','required_education'
pd.pivot_table(job_posting_small, index=['required_experience','required_education'])

fraudulent  ...  telecommuting
required_experience required_education                             ...               
Associate           Associate Degree                     0.000000  ...       0.000000
                    Bachelor's Degree                    0.000000  ...       0.000000
                    Certification                        0.000000  ...       0.000000
                    High School or equivalent            0.000000  ...       0.000000
                    Master's Degree                      0.000000  ...       0.000000
                    Unspecified                          0.000000  ...       0.000000
                    Vocational                           0.000000  ...       0.000000
Director            Associate Degree                     0.000000  ...       1.000000
                    Bachelor's Degree                    0.000000  ...       0.000000
                    Master's Degree                      0.000000  ...       0.000000
Entry level         Bachelor's Degree                    0.000000  ...       0.045455
                    High School or equivalent            0.012658  ...       0.012658
                    Some College Coursework Completed    0.000000  ...       0.000000
                    Unspecified                          0.000000  ...       0.000000
Executive           Bachelor's Degree                    0.000000  ...       0.000000
                    Unspecified                          0.000000  ...       0.000000
Internship          Bachelor's Degree                    0.000000  ...       0.333333
                    High School or equivalent            0.000000  ...       0.000000
                    Master's Degree                      0.000000  ...       0.000000
                    Some College Coursework Completed    0.000000  ...       0.000000
Mid-Senior level    Bachelor's Degree                    0.000000  ...       0.019608
                    Certification                        0.500000  ...       0.000000
                    High School or equivalent            0.500000  ...       0.000000
                    Master's Degree                      0.000000  ...       0.000000
                    Unspecified                          0.000000  ...       0.000000
Not Applicable      Bachelor's Degree                    0.000000  ...       0.000000
                    High School or equivalent            0.000000  ...       0.000000
                    Master's Degree                      0.000000  ...       0.000000
                    Unspecified                          0.000000  ...       0.000000

[29 rows x 5 columns]

In [ ]:
req_ex = job_posting_small.groupby('required_education').sum()
req_ex.describe()

job_id  telecommuting  has_company_logo  has_questions  fraudulent
count      8.000000        8.00000           8.00000       8.000000    8.000000
mean   10317.500000        0.62500          37.37500      19.375000    0.500000
std    16538.439285        1.06066          58.45618      36.901171    1.069045
min      172.000000        0.00000           1.00000       0.000000    0.000000
25%      709.250000        0.00000           1.75000       1.750000    0.000000
50%     1747.500000        0.00000           6.50000       4.500000    0.000000
75%    11623.250000        1.00000          45.00000      15.500000    0.250000
max    45823.000000        3.00000         158.00000     109.000000    3.000000

In [ ]:
job_posting_small[job_posting_small['required_education'].isin(['Associate Degree'])]



job_id            title  ... function fraudulent
366     367   Sales Director  ...    Sales          0
472     473  Account Manager  ...    Sales          0

[2 rows x 18 columns]